File: tools/conf/output/ts/RoomConfig.ts
class RoomConfig {
    private static _dic:Dictionary<RoomConfig> = new Dictionary<RoomConfig>();
    private static _arr:Array<RoomConfig> = new Array<RoomConfig>();

	 public Id :number ;
     public TotalType :number ;
     public Name :string ;
     public CompetitionIds : Array<number>  =new Array<number>();
     public StockName :string ;
     public IsOpen :number ;
     public DayType :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<RoomConfig>{
        return  RoomConfig._dic;
    }
    public static get Arr():Array<RoomConfig>{
        return RoomConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:RoomConfig = <RoomConfig>json[i];
            RoomConfig._arr.push(config);
            RoomConfig._dic.add(config.Id,config);
        }
    }
}

File: tools/conf/output/go/RoomConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type RoomConfigElem struct {
	Id int
	TotalType int
	Name string
	CompetitionIds []int
	StockName string
	IsOpen int
	DayType int
	
}

type RoomConfig0 struct {
	Dic map[int]*RoomConfigElem
	Arr []*RoomConfigElem
}

var (
	RoomConfig RoomConfig0
)

func loadRoomConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	RoomConfig.Dic = make(map[int]*RoomConfigElem)
	RoomConfig.Arr = make([]*RoomConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &RoomConfig.Arr)
	for _,conf := range RoomConfig.Arr{
		RoomConfig.Dic[conf.Id] = conf
	}
	
}
